sql/src/main/java/org/apache/druid/sql/avatica/DruidMeta.java

```diff
@@ -621,7 +613,12 @@ private DruidConnection getDruidConnection(final String connectionId)
     final DruidConnection connection = connections.get(connectionId);
 
     if (connection == null) {
-      throw logFailure(new NoSuchConnectionException(connectionId));
+      NoSuchConnectionException noSuchConnectionException = new NoSuchConnectionException(connectionId);
+      // this is to avoid an unecessary cast of NoSuchConnectionException to a runtime exception.
+      if (errorHandler.hasAffectingErrorResponseTransformStrategy()) {
```

Comment:
Is it possible to have this logic inside ErrorHandler?

Reply:
I can't put that inside the `ErrorHandler` because of the method's typing where everything becomes a runtime exception even if there is no error response strategy.
I can out a specific method that preserves this type for the `ErrorHandler`

Reply:
i'll add more docs to explain this

Reply:
Ah right.